model/src/main/kotlin/config/PackageConfiguration.kt

```diff
@@ -82,32 +83,37 @@ data class PackageConfiguration(
         }
     }
 
-    fun matches(otherId: Identifier, provenance: Provenance): Boolean {
-        @Suppress("ComplexCondition")
-        if (!id.type.equals(otherId.type, ignoreCase = true) ||
-            id.namespace != otherId.namespace ||
-            id.name != otherId.name ||
-            id.version != otherId.version
-        ) {
-            return false
+    private fun Identifier.matches(otherId: Identifier, supportVersionRange: Boolean): Boolean {
```

Comment:
Could `supportVersionRange` be omit and always the version range be compared to reduce complexity?
(This would also avoid the need to split-up this `matches()`))

Reply:
I can do the change.

However in the original spec you wrote, comparing with version range was limited to three cases:
https://github.com/oss-review-toolkit/ort/issues/9918#issuecomment-2696963351

So you confirm it should always be enabled ?

Reply:
Ah, ok I missed that. I' think about it again and the add another comment.